src/framework/components/particlesystem/particlesystem_system.js

```diff
@@ -195,12 +195,49 @@ pc.extend(pc, function() {
                 type: 'boolean',
                 defaultValue: false
             }, {
-                name: "spawnBounds",
-                displayName: "Spawn Bounds",
-                description: "Defines an AABB in which particles are allowed to spawn",
+                name: "emitterShape",
+                displayName: "Emitter Shape",
+                description: "Defines a shape in which particles are allowed to spawn",
+                type: "enumeration",
+                options: {
+                    enumerations: [{
+                        name: 'Box',
+                        value: pc.EMITTERSHAPE_BOX
+                    }, {
+                        name: 'Sphere',
+                        value: pc.EMITTERSHAPE_SPHERE
+                    }]
+                },
+                defaultValue: pc.EMITTERSHAPE_BOX,
+            }, {
+                name: "emitterExtents",
```

Comment:
This needs to be filtered on emitterShape. Check out collision_system.js to see how halfExtents is filtered on type.